framework/test_selectors/SelectorFactory.cpp

```diff
@@ -188,17 +189,51 @@ struct test *TestrunSelector::testid_to_test(const char *id, bool silent)
     return nullptr;
 }
 
+namespace {
+static constexpr auto arch_spr = "spr";
```

Comment:
from CodingGuidelines:
Constants and defines should be ALL_CAPS.

Reply:
I'd prefer we use `ALL_CAPS` only for macros. This is not one.